internal/checker.go

```diff
@@ -0,0 +1,125 @@
+package internal
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/go-resty/resty/v2"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+// Checker defines error based sqli checker for bWAPP.
+type Checker interface {
+	BWAPPAuth() Checker
+	CountErrs(body []byte) int
+	Check()
+}
+
+type checker struct {
+	url     string
+	payload string
+	errors  []string
+	client  *resty.Client
+}
+
+func NewChecker(url, payload string, errors []string, client *resty.Client) Checker {
+	c := &checker{
+		url:     url,
+		payload: payload,
+		errors:  errors,
+		client:  client,
+	}
+
+	return c
+}
+
+// BWAPPAuth gets valid cookies for bWAPP-site.
+func (c *checker) BWAPPAuth() Checker {
```

Comment:
Actually in most common cases your solution will be very nice. However this tight-forwarded authorization created just for the bWAPP-site which I run in a docker container. The idea is that this site doesn't return any error on login. It will be always 200 OK. So I chosen this approach returning an interface for having a possibility to make a chain function call then.